src/STEMMUS_SCOPE_exe.m

```diff
@@ -1,11 +1,63 @@
 %%%%%%% A function to run STEMMUS_SCOPE %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-function STEMMUS_SCOPE_exe(config_file)
+function STEMMUS_SCOPE_exe(config_file, run_mode)
+    if ~exist('run_mode', 'var')
+        runMode = "full";
+    else
+        runMode = run_mode;
+    end
 
     % disable all warnings
     warning('off', 'all');
 
-    % set the global variable CFG
+    % set the variable CFG
     CFG = config_file;
 
-    % run STEMMUS_SCOPE main code
-    run STEMMUS_SCOPE;
+    % If the runMode is "full" or was not provided, the model will run as normal
+    if strcmp(runMode, "full")
+        run STEMMUS_SCOPE;
+
+    elseif strcmp(runMode, "interactive")
+        % In interactive mode MATLAB stays open and waits for a new command...
+
+        % Define BMI required variable names:
+        bmiVarNames = {'ModelSettings', ... % Model settings struct
+                       'TimeStep', ... % Time step size (in seconds)
+                       'KT', ... % Index of current time step
+                       'SiteProperties', ... % Site properties (e.g. lat, lon)
+                       'fluxes', ... % Atmospheric fluxes
+                       'TT' ... % Soil temperature over depth
+                      }; %#ok
+
+        % ...until finalize has been run, at which point it quits.
+        while ~strcmp(runMode, "finalize")
+            runMode = input("\nFinished command. Select run mode: ", "s");
+
+            if strcmp(runMode, "initialize") || strcmp(runMode, "update") || strcmp(runMode, "finalize")
+                % The 'initialize', 'update' and 'finalize' run modes are dispatched to the model.
+                run STEMMUS_SCOPE;
+
+            elseif strcmp(runMode, "save")
+                % Save entire model state to file. This can be retrieved with the
+                %  model run mode "load", where this file is opened again.
+                save([Output_dir, 'STEMMUS_SCOPE_full_state.mat'], "-v7.3", "-nocompression");

```

Comment:
I am using Octave 7.3.0. The save function does not support arguments `"-v7.3", "-nocompression"`, can you please check if these are octave-compatible? see the latest [doc of octave](https://docs.octave.org/latest/Simple-File-I_002fO.html). This `save([Output_dir, 'STEMMUS_SCOPE_full_state.mat'], "-v7");` works though.

Reply:
I tried to use other formats, but I could not reload the workspace after opening and saving it in Python.

The 7.3 format is HDF5, which is easier to modify in-place in Python. However, Octave doesn't support it and Octave's hdf5 format is not compatible with Matlab.

It's difficult to support this for both Octave and Matlab, and seeing as the code runs a lot slower on Octave I feel like the BMI can be Matlab only.

Reply:
OK, then please add an inline comment above save functions here and in STEMMUS_SCOPE  about octave incompatibility. Can you also add a short description of BMI implementation in the main readme, thanks.